metis-indexing/src/main/java/eu/europeana/indexing/ClientsConnectionProvider.java

```diff
@@ -1,37 +1,42 @@
 package eu.europeana.indexing;
 
-import org.apache.solr.client.solrj.SolrClient;
 import eu.europeana.corelib.mongo.server.EdmMongoServer;
 import eu.europeana.indexing.exception.SetupRelatedIndexingException;
+import eu.europeana.metis.mongo.RecordRedirectDao;
+import org.apache.solr.client.solrj.SolrClient;
 
 /**
- * This class is an implementation of {@link AbstractConnectionProvider} that sets up the connection using
- * provided Solr and Mongo clients. Note: the caller is responsible for closing those connections.
- * 
- * @author jochen
+ * This class is an implementation of {@link AbstractConnectionProvider} that sets up the connection
+ * using provided Solr and Mongo clients. Note: the caller is responsible for closing those
+ * connections.
  *
+ * @author jochen
  */
 final class ClientsConnectionProvider implements AbstractConnectionProvider {
 
-  private final EdmMongoServer mongoClient;
+  private final EdmMongoServer edmMongoClient;
+  private final RecordRedirectDao recordRedirectDao;
   private final SolrClient solrClient;
 
   /**
    * Constructor.
-   * 
-   * @param mongoClient The Mongo client to be used. Cannot be null.
+   *
+   * @param edmMongoClient The Mongo client to be used. Cannot be null.
+   * @param recordRedirectDao The record redirect dao.
    * @param solrClient The Solr client to be used. Cannot be null.
    * @throws SetupRelatedIndexingException In case either of the two clients are null.
    */
-  ClientsConnectionProvider(EdmMongoServer mongoClient, SolrClient solrClient)
+  ClientsConnectionProvider(EdmMongoServer edmMongoClient, RecordRedirectDao recordRedirectDao,
+      SolrClient solrClient)
```

Comment:
Should you not do the null checks for the record redirect dao, just like for the other connections?

Reply:
Indeed.